### PR TITLE
Storage splitting and "List" enhancements

### PR DESCRIPTION
**Pull request by Roycius**
## Changelog

* Split storage into `tasks.txt`, `lessons.txt` and `modules.txt`
* Change command executions to work with new storage
* Add new exception catching for `taskList` and `lessonList` that deals with extra empty lines
* Add some JUnit for storage
* Add "today" and "tomorrow" to listing methods

### CODE REVIEW

File: src/main/java/seedu/duke/Duke.java [hunk 1]
```diff
@@ -25,8 +26,9 @@ public Duke() {
         ui = new Ui();
         storage = new Storage();
         try {
-            taskList = new TaskList(TaskList.deserialize(storage.loadData()));
-            lessonList = new LessonList(LessonList.deserialize(storage.loadData()));
+            taskList = new TaskList(TaskList.deserialize(storage.loadData("tasks.txt")));
+            lessonList = new LessonList(LessonList.deserialize(storage.loadData("lessons.txt")));
```
Comment (ptejasv) [inline src/main/java/seedu/duke/Duke.java:30]: Do you want to make "tasks.txt" and "lessons.txt" global constants instead, or maybe public constants in `Storage.java`? So if they're public this line would look like `taskList = new TaskList(TaskList.deserialize(storage.loadData(Storage.LESSON_FILE_NAME)));`. Same thing for the delete commands.

File: src/main/java/seedu/duke/storage/Storage.java [hunk 2]
```diff
@@ -14,14 +14,19 @@
...
 
 // Code reuse from https://github.com/richwill28/ip/blob/master/src/main/java/duke/storage/Storage.java
 public class Storage {
     private static final String ROOT_DIRECTORY = System.getProperty("user.dir");
     private static final String STORAGE_DIRECTORY = "data";
-    private static final String FILE_NAME = "duke.txt";
-    private static final Path PATH_TO_FILE = Paths.get(ROOT_DIRECTORY, STORAGE_DIRECTORY, FILE_NAME);
+    private static final String TASK_FILE_NAME = "tasks.txt";
+    private static final String LESSON_FILE_NAME = "lessons.txt";
+    private static final String MODULE_FILE_NAME = "modules.txt";
```
Comment (ptejasv) [inline src/main/java/seedu/duke/storage/Storage.java:26]: As per my previous suggestion, these would be public.

**Review by ptejasv — APPROVED** [review]
Looks good, just one small suggestion!

File: src/main/java/seedu/duke/Duke.java (hunk 1 above)
Comment (Roycius) [reply src/main/java/seedu/duke/Duke.java:30]: Good point! I will try to make them public constants in `Storage.java`.

**Pull request merged**